sdks/python/apache_beam/io/filebasedsource_test.py

```diff
@@ -41,6 +41,14 @@
 class LineSource(FileBasedSource):
 
   def read_records(self, file_name, range_tracker):
+    current = -1
+
+    def _callback(stop_position):
+      return FileBasedSource.remaining_split_points_helper(
+          stop_position, current, range_tracker.done())
```

Comment:
why do you not need a progress lock here? there are two calls made to things that may be changed by the read loop: `current` and `range_tracker`.